packages/mermaid/src/diagrams/git/gitGraphRenderer.js

```diff
@@ -338,27 +338,24 @@ const drawCommits = (svg, commits, modifyGraph) => {
 };
 
 /**
- * Detect if there are other commits between commit1's x-position and commit2's x-position on the
- * same branch as commit2.
+ * Detect if there are commits
+ * between commitA's x-position
+ * and commitB's x-position on the
+ * same branch as commitA
  *
- * @param {any} commit1
- * @param {any} commit2
+ * @param {any} commitA
+ * @param {any} commitB
  * @param allCommits
- * @returns {boolean} If there are commits between commit1's x-position and commit2's x-position
+ * @returns {boolean}
+ * If there are commits between
+ * commitA's x-position
+ * and commitB's x-position
  */
-const hasOverlappingCommits = (commit1, commit2, allCommits) => {
-  // Find commits on the same branch as commit2
-  const keys = Object.keys(allCommits);
-  const overlappingComits = keys.filter((key) => {
-    return (
-      allCommits[key].branch === commit2.branch &&
-      allCommits[key].seq > commit1.seq &&
-      allCommits[key].seq < commit2.seq
-    );
-  });
-
-  return overlappingComits.length > 0;
-};
+const hasOverlappingCommits = (commitA, commitB, allCommits) =>
+  Object.entries(allCommits).some(
+    (commitX) =>
+      commitX.branch === commitA.branch && commitX.seq > commitA.seq && commitX.seq < commitB.seq
```

Comment:
@mathbraga Thanks for the review of this. It took me a while to understand the original code (which is partly why I rewrote it to try and make it clearer, and marginally more performant).

`hasOverlappingCommits` should return true when two commits that are being connected have other commits between them on the source branch (i.e, the same branch as `commitA` in my rewrite). This being the case, the arrow's path then needs to be diverted so as to avoid running over the other commits and making the graph unclear.

Hope this makes sense but do keep asking questions if you have any, as it may be that I've missed something.

Reply:
Yes it makes sense. I was unaware of how the cherry-pick feature works exactly, and that is one of the situations where this case would occur.

(the code you provided in the new demos, with the above proposed change in `hasOverlappingCommits`)
![image](https://github.com/mermaid-js/mermaid/assets/31048764/c8b39c9e-1d77-49a4-b385-a82b8f138e2d)

Sorry for the confusion.

Reply:
Could you resolve this conversation if you are happy with the outcome? Thanks!